### ISSUE TITLE
[QUESTION] Using a background class subclass

### ISSUE THREAD

**Issue by kevgliss** [tiangolo/fastapi#805]
### First check

* [X] I used the GitHub search to find a similar issue and didn't find it.
* [X] I searched the FastAPI documentation, with the integrated search.
* [X] I already searched in Google "How to X in FastAPI" and didn't find any information.

### Description

What's the best way to use a Background task subclass? As far as I can tell FastAPI will use starlette.background.BackgroundTasks no matter what. I was hoping that by defining my params as:

`background_tasks: MySubClassBackgroundTasks,`

 I could get FastAPI to use my subclass. But it seems that only does some pydantic type checking for dependencies, but does not allow me to specify my class.

Is the best way to fall back to starlette itself? Attaching the tasks to a response like mentioned here: 
https://www.starlette.io/background/

**Comment by tiangolo** [comment]
Yes, the declared background tasks are used only to add functions to a list, but that instance is not even used, those functions are added to the actual background class. So yeah, declaring a custom background class won't work.

And yes, if you return a response directly, you can add your background class there, I don't remember exactly the code flow, but I'm pretty sure that should work.

On the other side, I can't imagine, why would you need a custom background class?

**Comment by kevgliss** [comment]
The use case for a custom background class is to add custom exception handling (sentry) and metrics gathering to all background tasks -- Is there be a better way to handle that use case?

**Comment by tiangolo** [comment]
That's an interesting point and use case. I'm still not sure what would be the best approach to solve it :thinking:
